spec/lib/license_finder/package_managers/pip_spec.rb

```diff
@@ -55,11 +56,28 @@ module LicenseFinder
         let(:pip) { Pip.new(project_path: Pathname(root), pip_requirements_path: @user_provided_requirements) }
 
         it 'should use the provided requirements file' do
-          expect(SharedHelpers::Cmd).to receive(:run).with("pip install -r #{@user_provided_requirements}")
+          expect(SharedHelpers::Cmd).to receive(:run).with("pip2 install -r #{@user_provided_requirements}")
                                                      .and_return([dependency_json, '', cmd_success])
           pip.prepare
         end
       end
+
+      context 'pip configured with python 3' do
+        let(:pip) { Pip.new(project_path: Pathname(root), pip_requirements_path: @user_provided_requirements, python_version: '3') }
+
+        it 'should call the pip3 binary' do
+          expect(SharedHelpers::Cmd).to receive(:run).with("pip3 install -r #{@user_provided_requirements}")
+                                                     .and_return([dependency_json, '', cmd_success])
+          pip.prepare
+        end
+      end
+
+      context 'pip configured with a unknown python version' do
+        it 'should error out' do
+          expect { Pip.new(project_path: Pathname(root), pip_requirements_path: @user_provided_requirements, python_version: '100') }
+          .to raise_error("Invalid python version '100'. Valid versions are '2' or '3'.")
+        end
+      end
     end
```

Comment:
Maybe split this test into 2 separate `Context`, one for `python 2` and one for `python 3`. This may help organize the test better